examples/shader/texture_binding_array.rs

```diff
@@ -47,7 +32,21 @@ fn setup(
     mut meshes: ResMut<Assets<Mesh>>,
     mut materials: ResMut<Assets<BindlessMaterial>>,
     asset_server: Res<AssetServer>,
+    render_device: Res<RenderDevice>,
 ) {
+    // check if the device support the required feature
```

Comment:
If I recall correctly, this was in main partially as a way of demonstrating graceful handling of feature detection.

But now with wasm/webgl2 at least, we get a panic before this code gets a chance to run.

Is this code even useful anymore?

```
sm.js:387 panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_bind_group_layout
      note: label = `bindless_material_layout`
    Binding 0 entry is invalid
    Features Features(TEXTURE_BINDING_ARRAY) are required but not enabled on the device

', /Users/me/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.16.0/src/backend/direct.rs:3019:5
```

Reply:
That should really be moved to main, after the default plugin is setup and before the custom material is setup.

Reply:
> But now with wasm/webgl2 at least, we get a panic before this code gets a chance to run.

Oh that gives the correct log in wasm/WebGPU, but not in wasm/WebGL2...

> That should really be moved to main(), after the default plugin is setup and before the custom material is setup.

That's not possible anymore, the renderer is not available before the winit event loop is started.

Reply:
Is it only the device that needs to be created in finish()? Can we create a queue in the main plugin build, and use that to check for feature support?

Reply:
Getting the adapter is async: https://docs.rs/wgpu/latest/wgpu/struct.Instance.html#method.request_adapter
Getting the device is async: https://docs.rs/wgpu/latest/wgpu/struct.Adapter.html#method.request_device

Reply:
For context, I need this pattern of checking whether a feature is supported before loading a plugin for my own crates. It's important that we find a way to solve it nicely :). It's not something we can just let an example crash if not supported.

Reply:
A more complete fix would be to introduce a "fake" plugin in the example that just does the check, but nothing else from what's expected from a plugin

Reply:
@JMS55 would https://github.com/bevyengine/bevy/pull/8553/files#diff-86a9f1eba7a2e2bcb940a56d71acb7675c9c24ab6981ed7f221a10f9117514f0R32 work for you?

Reply:
I'm not sure. What I need is the ability to add a plugin only if certain features are supported. Here's an example: https://github.com/JMS55/bevy/blob/solari/crates/bevy_solari/src/lib.rs#L45-L62.

Reply:
That's not going to be possible like that, without putting all the app initialisation as async.

What you can do is add your plugins, but have them do nothing if the feature are not available.

Reply:
I don't love this solution, but I guess it'll have to do :(